server/src/main/java/io/crate/types/IntervalType.java

```diff
@@ -151,4 +153,20 @@ public int fixedSize() {
     public long valueBytes(Period value) {
         return 32;
     }
+
+    /**
+     * A copy of {@link Period#toStandardDuration()} which allows also months and years
+     * using a standard number of 30 days per month and 365 days per year.
+     */
+    public static Duration toStandardDuration(Period p) {
+        long millis = p.getMillis();  // no overflow can happen, even with Integer.MAX_VALUEs
+        millis += (((long) p.getSeconds()) * ((long) DateTimeConstants.MILLIS_PER_SECOND));
+        millis += (((long) p.getMinutes()) * ((long) DateTimeConstants.MILLIS_PER_MINUTE));
+        millis += (((long) p.getHours()) * ((long) DateTimeConstants.MILLIS_PER_HOUR));
+        millis += (((long) p.getDays()) * ((long) DateTimeConstants.MILLIS_PER_DAY));
+        millis += (((long) p.getWeeks()) * ((long) DateTimeConstants.MILLIS_PER_WEEK));
+        millis += (p.getMonths() * (DateTimeConstants.MILLIS_PER_DAY * 30L));
+        millis += (p.getYears() * (DateTimeConstants.MILLIS_PER_DAY * 365L));
```

Comment:
Doesn't this overflow if years or months is close to Integer.MAX_VALUE?

E.g.:

```
jshell> Integer.MAX_VALUE * (86400000L * 365L);
$1 ==> -6063932003046206464
```

Reply:
Right, but this can potentially happen with other fields, for example: `SELECT ts2 - ts1 FROM test ORDER BY 1`. Here with another PR we followed postgres behaviour: https://github.com/crate/crate/commit/bcc469274b63a5e52dcb668b18040aeba1253df3, so depending on the timestamps you can have a very large amount of days, which can also cause overflow.

Do you have any suggestions?

Reply:
We could normalise each value/unit

E.g. 2 hours, 72 seconds -> 2 hours, 1 minutes, 12 seconds

And then we can compare hour, minute, seconds, etc. individually. While we're at it we could also avoid the allocations and to the comparisons directly

Reply:
Let me try this, thx!